### PR TITLE
Passing url_type into template context.

### PR DESCRIPTION
**Pull request by felixfontein**
Also modifies base theme to use that when calling `url_replacer`. This allows to fix the Nikola core part of getnikola/plugins#176.

### CODE REVIEW

File: nikola/nikola.py [hunk 1]
```diff
@@ -1306,6 +1306,7 @@ def render_template(self, template_name, output_name, context, url_type=None):
         for k in self._GLOBAL_CONTEXT_TRANSLATABLE:
             local_context[k] = local_context[k](local_context['lang'])
         local_context['is_rtl'] = local_context['lang'] in LEGAL_VALUES['RTL_LANGUAGES']
+        local_context['url_type'] = self.config['URL_TYPE'] if url_type is None else url_type
```
Comment (Kwpolska) [inline nikola/nikola.py]: Now that we’re doing all this, perhaps we could pass `url_type` into this function from callers?

Comment (felixfontein) [inline nikola/nikola.py]: Why should callers pass the argument twice (both into context and to `render_template`)? After all this value is used in the default base theme in essentially every template.

Comment (felixfontein) [inline nikola/nikola.py]: I couldn't find a reason why anyone calling this function does NOT want to have `url_type` in context (except that it causes all pages to recompile, that is).

Comment (Kwpolska) [inline nikola/nikola.py]: Ah, `generic_` don’t use `url_type` themselves. Feel free to merge.

Comment (felixfontein) [inline nikola/nikola.py]: Ok, thanks, I'll merge then.

**Pull request merged**